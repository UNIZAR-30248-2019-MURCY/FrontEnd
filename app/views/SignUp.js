import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text,
    ScrollView
} from 'react-native'
import {Button, CheckBox } from 'react-native-elements';
import {signUpUser} from '../services/user/userFuncs';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class SignUp extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            password2: '',
            email: '',
            phone_number: '',
            checked: false,
            error: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if (this.state.username !== '' && this.state.password !== '' && this.state.email !== '') {
            signUpUser(this.state.username, this.state.password, this.state.email)
                .then((data) => {
                    console.log(data);
                    //this.setState(data);
                    this.props.history.replace('/emailconfirm');
                })
                .catch((error) => {
                    this.setState({error: error.message})
                })
        } else if (!this.state.checked) {
            this.setState({error: 'You must accept the terms and conditions '})
        } else if (this.state.password !== this.state.password2) {
            this.setState({error: 'Passwords must be the same '})
        } else {
            this.setState({error: 'Introduzca todos los campos '})
        }
    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error}>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        

        return (
            <View style={styles.container}>
                <ScrollView>
                <View style={styles.cross}>
                    <Button
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.history.push('/');
                        }
                        }
                    />
                </View>

                <View style={styles.login}>
                    <Image
                        style={styles.logo}
                        source={require('./../assets/images/murcy.png')}
                    />
                    <TextInput
                        style={styles.input}
                        placeholder='Username'
                        value={this.state.username}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('username', val)}
                    />
                    <TextInput
                        style={styles.input}
                        placeholder='Email'
                        keyboardType='email-address'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('email', val)}
                    />
                    <TextInput
                        style={styles.input}
                        placeholder='Password'
                        secureTextEntry={true}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('password', val)}
                    />
                    <TextInput
                        style={styles.input}
                        placeholder='Repeat password'
                        secureTextEntry={true}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('password', val)}
                    />
                    <CheckBox
                        center
                        title={<Text style={{color: 'blue', textDecorationLine: 'underline'}}
                        onPress={() => {
                            this.props.history.push('/termsConditions');
                        }}>I have read and I accept the Terms and Conditions </Text>}
                        checked={this.state.checked}
                        onPress={() => this.setState({ checked: !this.state.checked })}
                    />

                    {showErr}

                    <Button
                        buttonStyle={styles.button}
                        title="Sign Up"
                        onPress={() => {
                            this.handleSubmit()
                        }}/>


                    <Button
                        className='login-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Log In"
                        titleStyle={{color: 'grey'}}
                        onPress={() => {
                            this.props.history.replace('/login');
                        }
                        }/>
                </View>
                </ScrollView>
            </View>
        )
    }
}

const
    styles = StyleSheet.create({
        container: {
            flex: 1,
            //justifyContent: 'center',
        },
        cross: {
            marginTop: 50,
            marginBottom: 40,
            marginRight: 10,
            alignItems: 'flex-end',
        },
        login: {
            flex: 1,
            alignItems: 'center',
            //justifyContent: 'center',
        },
        logo: {
            marginBottom: 30,
            padding: 8,
            width: 100,
            height: 100
        },
        input: {
            width: 300,
            height: 55,
            backgroundColor: 'whitesmoke',
            margin: 10,
            padding: 8,
            color: 'grey',
            borderRadius: 14,
            fontSize: 18,
            fontWeight: '500',
        },
        button: {
            width: 150,
            height: 55,
            marginTop: 20,
            margin: 10,
            backgroundColor: 'grey',
            borderRadius: 14
        },
        button2: {
            width: 150,
            height: 55,
            borderRadius: 14,
            marginBottom: 50,
        },
        error: {
            margin: 10,

        },

    })
