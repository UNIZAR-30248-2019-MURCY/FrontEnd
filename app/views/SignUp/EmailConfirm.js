import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    Image,
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class EmailConfirm extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.cross}>
                    <Button
                        className='close-button'
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.navigation.navigate('AuthLoading');
                        }
                        }
                    />
                </View>

                <View style={styles.containerMessage}>
                    <Icon
                        className='confirm-email'
                        name='check'
                        size={100}
                    />
                    <Text style={{color: 'grey', fontSize: 19, marginTop:20, textAlign:'center'}} >
                        Check your inbox to verify your email
                    </Text>
                </View>

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,

    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    containerMessage: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 100,
        padding: 20
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    button: {
        marginTop: 70,
        width: 150,
        height: 55,
        padding: 8,
    },
    buttonText: {
        fontSize: 25,
        color: 'grey'
    }

})
