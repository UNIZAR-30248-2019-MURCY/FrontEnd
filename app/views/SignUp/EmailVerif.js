import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    ActivityIndicator,
    Linking
} from 'react-native'
import {Button, Text} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {emailVerif} from "../../services/user/userFuncs";
import WEB from '../../config/web';


export default class EmailVerif extends Component {

    constructor(props) {
        super(props);
        this.state = {
            loading : true,
            error: false,
        };

    }
    componentDidMount() {
        if (this.props.navigation) {
            if(this.props.navigation.getParam('token')){
                emailVerif(this.props.navigation.getParam('token'))
                    .then((data) => {
                        this.setState({loading: false})
                    })
                    .catch((error) => {
                        this.setState({error: error.message}),
                            this.setState({loading: false})
                    })
            }
            else{
                this.props.navigation.navigate('AuthLoading');
            }
        }
    }

    render() {
        let showConfirm = (
            this.state.error ?
                <View  style={styles.containerMessage} className='errorShow'>
                        <Icon
                            name='times'
                            size={100}
                        />
                        <Text style={{color: 'grey', fontSize: 19, marginTop:20, textAlign:'center'}} >
                            {this.state.error}
                        </Text>
                        <Button
                            className='return-button'
                            buttonStyle={styles.button}
                            title="Home"
                            onPress={() => {
                                Linking.openURL(WEB.URL)
                            }}/>
                </View> :
                    <View style={styles.containerMessage}>
                        <Icon
                            name='check'
                            size={100}
                        />
                        <Text style={{color: 'grey', fontSize: 19, marginTop:20, textAlign:'center'}} >
                            Your email has been successfully verified
                        </Text>
                        <Button
                            className='return-button'
                            buttonStyle={styles.button}
                            title="Home"
                            onPress={() => {
                                Linking.openURL(WEB.URL)
                            }}/>
                    </View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerMessage, styles.horizontal]}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View>
                    {showConfirm}
                </View>
        );

        return (
            <View style={styles.container}>
                {showLoading}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,

    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    containerMessage: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 180,
        padding: 20
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    button: {
        width: 120,
        height: 40,
        marginTop: 90,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    buttonText: {
        fontSize: 25,
        color: 'grey'
    }

})
