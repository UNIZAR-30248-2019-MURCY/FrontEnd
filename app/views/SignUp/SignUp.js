import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text,
    ScrollView, ActivityIndicator
} from 'react-native'
import {Button, CheckBox} from 'react-native-elements';
import {signUpUser} from '../../services/user/userFuncs';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class SignUp extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            password2: '',
            email: '',
            fullName: '',
            checked: false,
            error: false,
            loading: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }


    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if (this.state.username !== '' && this.state.password !== '' && this.state.email !== '' && this.state.fullName !== '') {
            if (this.state.checked) {
                if (this.state.password === this.state.password2) {
                    this.setState({loading: true})
                    this.setState({error: false})
                    signUpUser(this.state.username, this.state.password, this.state.email, this.state.fullName)
                        .then((data) => {
                            this.setState({loading: false}),
                                this.props.navigation.replace('EmailConfirm');
                        })
                        .catch((error) => {
                            this.setState({error: error.message}),
                                this.setState({loading: false})
                        })
                } else {
                    this.setState({error: 'Passwords must be the same '})
                }
            } else {
                this.setState({error: 'You must accept the terms and conditions '})
            }
        } else {
            this.setState({error: 'Introduzca todos los campos '})
        }
    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading, styles.horizontal]}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (
            <View style={styles.container}>
                <ScrollView>
                    <View style={styles.cross}>
                        <Button
                            className='close-button'
                            type="clear"
                            icon={
                                <Icon
                                    name="times"
                                    size={30}
                                    color="grey"
                                />
                            }
                            onPress={() => {
                                this.props.navigation.navigate('AuthLoading')
                            }
                            }
                        />
                    </View>

                    <View style={styles.login}>
                        <Image
                            style={styles.logo}
                            source={require('../../assets/images/murcy.png')}
                        />
                        <TextInput
                            className='userInput'
                            style={styles.input}
                            placeholder='Username'
                            value={this.state.username}
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('username', val)}
                        />
                        <TextInput
                            className='emailInput'
                            style={styles.input}
                            placeholder='Email'
                            keyboardType='email-address'
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('email', val)}
                        />
                        <TextInput
                            className='passInput'
                            style={styles.input}
                            placeholder='Password'
                            secureTextEntry={true}
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('password', val)}
                        />
                        <TextInput
                            className='passrepInput'
                            style={styles.input}
                            placeholder='Repeat password'
                            secureTextEntry={true}
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('password2', val)}
                        />
                        <TextInput
                            className='nameInput'
                            style={styles.input}
                            placeholder='Full Name'
                            value={this.state.fullName}
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('fullName', val)}
                        />
                        <CheckBox
                            center
                            containerStyle={styles.checkBox}
                            title={<Text style={{color: 'black'}}
                                         onPress={() => {
                                             this.props.navigation.navigate('TermsConditions');
                                         }}>I have read and I accept the Terms and Conditions </Text>}
                            checked={this.state.checked}
                            onPress={() => this.setState({checked: !this.state.checked})}
                        />

                        {showErr}
                        {showLoading}
                        <Button
                            className='signup-button'
                            buttonStyle={styles.button}
                            title="Sign Up"
                            onPress={() => {
                                this.handleSubmit()
                            }}/>


                        <Button
                            className='login-button'
                            type="clear"
                            buttonStyle={styles.button2}
                            title="Log In"
                            titleStyle={{color: 'grey'}}
                            onPress={() => {
                                this.props.navigation.replace('LogIn');
                            }
                            }/>
                    </View>

                </ScrollView>
            </View>
        )
    }
}

const
    styles = StyleSheet.create({
        container: {
            flex: 1,
            //justifyContent: 'center',
        },
        containerLoading: {
            flex: 1,
            justifyContent: 'center'
        },
        horizontal: {
            flexDirection: 'row',
            justifyContent: 'space-around',
            padding: 10
        },
        cross: {
            marginTop: 50,
            marginBottom: 40,
            marginRight: 10,
            alignItems: 'flex-end',
        },
        login: {
            flex: 1,
            alignItems: 'center',
            //justifyContent: 'center',
        },
        logo: {
            marginBottom: 30,
            padding: 8,
            width: 100,
            height: 100
        },
        input: {
            width: 300,
            height: 55,
            backgroundColor: 'whitesmoke',
            margin: 10,
            padding: 8,
            color: 'grey',
            borderRadius: 14,
            fontSize: 18,
            fontWeight: '500',
        },
        button: {
            width: 150,
            height: 55,
            marginTop: 20,
            margin: 10,
            backgroundColor: 'grey',
            borderRadius: 14
        },
        button2: {
            width: 150,
            height: 55,
            borderRadius: 14,
            marginBottom: 50,
        },
        error: {
            margin: 10,

        },
        checkBox: {
            width: 300,
            marginTop: 20,
            borderRadius: 14,
        },

    })
