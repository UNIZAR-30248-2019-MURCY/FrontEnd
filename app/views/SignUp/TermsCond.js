import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    ScrollView
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';

export default class Terms extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <ScrollView>
                <View style={styles.containerTitle}>
                    <Text h2>Terms and conditions</Text>
                </View>

                <View style={styles.containerMurcy}>
                    <View>
                        <Text> These Terms and Conditions regulate the download, access and use of the MURCY application.</Text>
                        <Text> Access to the MURCY assumes that the user acknowledges that he has accepted and consented
                            without reservation of the present conditions of use.</Text>
                        <Text h5>1. OBJECTIVE </Text>
                        <Text> The MURCY application has the objective of entertainment.</Text>
                        <Text> It is an application for all audiences. In the design
                            and development of this application have participated specialized professionals.</Text>
                        <Text> The MURCY is made available to users for personal (never business) use.</Text>
                        <Text> It works on Web and an Android and iOS mobile phone.</Text>
                        <Text h5>2. INTELLECTUAL AND INDUSTRIAL PROPERTY RIGHTS </Text>
                        <Text> The user acknowledges that the reproduction, modification, distribution, commercialization, decompilation,
                             disassembly, use of reverse engineering techniques or any other means to obtain the source code,
                             transformation or publication of any results of tests of unauthorized references of any of the elements
                             and utilities integrated within the development constitutes an infringement of the intellectual
                             property rights of MURCY, being obliged, consequently, not to perform any of the aforementioned actions.</Text>
                        <Text h5>3. PRIVACY POLICY </Text>
                        <Text> MURCY may use personal information for internal purposes, such as statistics.
                            MURCY may collect, store or accumulate certain non-personal information regarding its use.</Text>
                        <Text> MURCY reserves the right to edit, update, modify, suspend, delete or terminate the services offered by
                            the Application, including all or part of its content, without prior notice, as well as to modify the form
                            or type of access to it. The possible causes of modification may take place, for reasons such as its
                            adaptation to possible legislative developments and changes in the Application itself, as well as to
                            those that may derive from the existing standard codes in the matter or for strategic or corporate reasons .</Text>
                        <Text> MURCY will not be responsible for the use of the application by a minor, being the download and use of
                            the application of the exclusive responsibility of the user.</Text>
                        </View>
                    <Button
                        className='back-button'
                        type="clear"
                        buttonStyle={styles.button}
                        title="Return"
                        titleStyle={styles.buttonText}
                        onPress={() => {
                            this.props.navigation.replace('SignUp');
                        }}/>
                </View>
                </ScrollView>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerMurcy: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 10,
        padding: 20
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    button: {
        marginTop: 40,
        width: 150,
        height: 55,
        padding: 8,
    },
    buttonText: {
        fontSize: 25,
        color: 'grey'
    }

})
