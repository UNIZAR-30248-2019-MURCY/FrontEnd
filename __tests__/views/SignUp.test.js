import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SignUp from "../../app/views/SignUp";

describe('<SignUp />', () => {
    it('SignUp renders without crashing', () => {
        const rendered = renderer.create(<SignUp />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('SignUp test against snapshot', () => {
        const welcomePage = renderer.create(<SignUp />).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('should navigate to Log In', () => {
        const historyMock = { replace: jest.fn() };
        const wrapper = shallow(<SignUp history={historyMock} />);
        wrapper.find('.login-button').simulate('press');
        expect(historyMock.replace.mock.calls.length).toEqual(1);
        expect(historyMock.replace.mock.calls[0]).toEqual(['/login']);
    })
});
