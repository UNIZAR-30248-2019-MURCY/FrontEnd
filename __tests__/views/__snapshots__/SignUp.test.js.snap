// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> SignUp test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "marginBottom": 40,
            "marginRight": 10,
            "marginTop": 50,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 100,
              "marginBottom": 30,
              "padding": 8,
              "width": 100,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Repeat password"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Full Name"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fafafa",
              "borderColor": "#ededed",
              "borderRadius": 14,
              "borderWidth": 1,
              "margin": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 20,
              "opacity": 1,
              "padding": 10,
              "width": 300,
            }
          }
          testID="checkbox"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text />
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "black",
                }
              }
            >
              I have read and I accept the Terms and Conditions 
            </Text>
          </View>
        </View>
        <View />
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "grey",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 55,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginTop": 20,
                  "padding": 8,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Sign Up
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 55,
                  "justifyContent": "center",
                  "marginBottom": 50,
                  "padding": 8,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Log In
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
